### ISSUE TITLE
Any reason we don't have CounterVecFunc & GaugeVecFunc?

### ISSUE THREAD

**Issue by sazary** [prometheus/client_golang#997]
Hi  

I noticed that this package provides, for example, `GaugeVec` and `GaugeFunc`, but it doesn't provide anything like `GaugeVecFunc`, which would gather a bunch of gauges whose value would be determined at collect time.  

I think I need something like this in my use case, and since it's not provided in this package I started to implement it myself. But then I wondered whether there is a reason that you've not implemented this, like maybe it's somehow a bad practice or it shows a problem in my metrics design.

So, is there any reason?

Thanks

**Comment by aureliar8** [comment]
This discussion seems to answer your question: https://github.com/prometheus/client_golang/discussions/924

TLDR: You can create a custom Collector to get the same functionality and the package already has a lot / too much exported types.

**Comment by beorn7** [comment]
Yeah, https://github.com/prometheus/client_golang/discussions/924 is precisely the right place for this discussion. Thank you.
